code/game/communications.dm

```diff
@@ -173,15 +173,19 @@ GLOBAL_LIST_INIT(reverseradiochannels, list(
 				if(start_point.get_virtual_z_level() != end_point.get_virtual_z_level() || (range > 0 && get_dist(start_point, end_point) > range))
 					continue
 			device.receive_signal(signal)
+			CHECK_TICK
 
 /datum/radio_frequency/proc/add_listener(obj/device, filter as text|null)
 	if (!filter)
 		filter = "_default"
 
+	var/datum/weakref/new_listener = WEAKREF(device)
+	if(isnull(new_listener))
+		return stack_trace("null, non-datum, or qdeleted device")
```

Comment:
This is not how stack trace works, stack trace does not return.
Stack trace continues the code execution, CRASH() halts it. This should be CRASH instead and shouldn't return, since crash returns for you